spring-integration-smb/src/test/java/org/springframework/integration/smb/config/SmbInboundChannelAdapterParserTests.java

```diff
@@ -75,14 +74,14 @@ public void testSmbInboundChannelAdapterComplete() throws Exception {
 		String remoteFileSeparator = (String) TestUtils.getPropertyValue(fisync, "remoteFileSeparator");
 		assertNotNull(remoteFileSeparator);
 		assertEquals("", remoteFileSeparator);
-		SmbSimplePatternFileListFilter filter = (SmbSimplePatternFileListFilter) TestUtils.getPropertyValue(fisync, "filter");
-		assertNotNull(filter);
+		//SmbSimplePatternFileListFilter filter = (SmbSimplePatternFileListFilter) TestUtils.getPropertyValue(fisync, "filter");
```

Comment:
@artembilan I assume this behavior change is due to Spring integration changes. If I am wrong kindly, guide me how can revert this.

Reply:
Yeah... That's right: starting with some version (https://jira.spring.io/browse/INT-4115), we populate there a `CompositeFileListFilter`. 
You should do something like this here:
```
FileListFilter<?> filter = TestUtils.getPropertyValue(fisync, "filter", FileListFilter.class);
assertNotNull(filter);
assertThat(filter, instanceOf(CompositeFileListFilter.class));
Set<?> fileFilters = TestUtils.getPropertyValue(filter, "fileFilters", Set.class);

Iterator<?> filtersIterator = fileFilters.iterator();
assertThat(filtersIterator.next(), instanceOf(SmbSimplePatternFileListFilter.class));
assertThat(filtersIterator.next(), instanceOf(SmbPersistentAcceptOnceFileListFilter.class));
```

Reply:
Thanks, I will get this updated.